services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightProducer.java

```diff
@@ -256,21 +259,42 @@ public void closePreparedStatement(
   public FlightInfo getFlightInfoStatement(
     CommandStatementQuery commandStatementQuery,
     CallContext callContext, FlightDescriptor flightDescriptor) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("Statement not supported.").toRuntimeException();
+    final UserSession session = getUserSessionFromCallContext(callContext);
+
+    final FlightPreparedStatement flightPreparedStatement = flightWorkManager
+      .createPreparedStatement(commandStatementQuery.getQuery(), callContext::isCancelled, session);
+
+    flightPreparedStatementCache.put(flightPreparedStatement.getServerHandle(), flightPreparedStatement);
+
+    TicketStatementQuery ticket =
+      TicketStatementQuery.newBuilder()
+        .setStatementHandle(flightPreparedStatement.getServerHandle().toByteString())
+        .build();
+
+    Schema schema = flightPreparedStatement.getSchema();
+    return getFlightInfoForFlightSqlCommands(ticket, flightDescriptor, schema);
   }
 
   @Override
   public SchemaResult getSchemaStatement(
     CommandStatementQuery commandStatementQuery,
     CallContext callContext, FlightDescriptor flightDescriptor) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("Statement not supported.").toRuntimeException();
+    FlightInfo info = this.getFlightInfo(callContext, flightDescriptor);
+    return new SchemaResult(info.getSchema());
   }
 
   @Override
-  public void getStreamStatement(CommandStatementQuery commandStatementQuery,
-                                 CallContext callContext, Ticket ticket,
+  public void getStreamStatement(TicketStatementQuery commandStatementQuery,
+                                 CallContext callContext,
```

Comment:
What happened to the `Ticket ticket` parameter?

Reply:
Fixed - that was missing on the current flight-sql version I had built